splash/browser_tab.py

```diff
@@ -892,6 +907,12 @@ def _send_request(self, url, callback, method='GET', body=None,
 
         request = self.request_obj(url, headers=headers, body=body)
 
+        # setting UA for request that is not downloaded via webpage.mainFrame().load_to_mainframe()
+        ua_from_headers = lowercase_byte_dict(headers).get(b"user-agent")
+        web_page_ua = self.web_page.userAgentForUrl(to_qurl(url))
+        user_agent = ua_from_headers if ua_from_headers else web_page_ua
+        request.setRawHeader(b"user-agent", to_bytes(user_agent))
```

Comment:
ammeded previous commit @kmike 

* not converting values to bytes
* not copying headers twice
* removed copy() call from utils function